eng/common/pipelines/templates/jobs/eng-common-workflow-enforcer.yml

```diff
@@ -0,0 +1,27 @@
+trigger: none
+
+pr:
+  branches:
+    include:
+      - master
+  paths:
+    include:
+      - eng/common
+
+jobs:
+  - job: EnforceCommonWorkflow
+
+    pool:
+      vmImage: windows-2019
+
+    steps:
+      - pwsh: |
+          $PrSourceBranch = $(System.PullRequest.SourceBranch)
+          Write-Host "[debug]Pull Request Source Branch: ${PrSourceBranch}"
+          if (-not $PrSourceBranch.StartsWith("sync-eng/common"))
```

Comment:
You should be able to produce a diff of the files changed and if there is anything under eng/common then enable this check.